ghzdac/calibrate.py

```diff
@@ -274,6 +274,72 @@ def measureImpulseResponse(fpga, scope, baseline, pulse, dacoffsettime=6, pulsel
     return (data)
 
 
+def measureImpulseResponse_infiniium(fpga, scope, baseline, pulse,
+                                     dacoffsettime=6, pulselength=1, wait=50, looplength=6000):
+    """Measure the response to a DAC pulse
+    looplength: time between triggers, keep this above pulselength , 6000 for short dacs
+    fpga: connected fpga server
+    scope: connected scope server
+    dac: 'a' or 'b'
+    returns: list
+    list[0] : start time (s)
+    list[1] : time step (s)
+    list[2:]: actual data (V)
+    """
+    #units clock cycles
+    dacoffsettime = int(round(dacoffsettime))
+    triggerdelay = 30 #keep at least at 30
+    pulseindex = triggerdelay-dacoffsettime
+
+    data = np.resize(baseline, looplength)
+    data[pulseindex:pulseindex+pulselength] = pulse
+    data[0] |= trigger
+    fpga.dac_run_sram(data,True)
+    if wait:
+        time.sleep(wait) #keep this long enough!! 40 sec for 4096, 20 sec for 2048
+
+    [timeAxis,y] = (scope.get_trace(SCOPECHANNEL_infiniium)) #start and stop in ns
+    timeAxis = [t*1.0e9 for t in timeAxis] #Agilent returns it in s, here we make ns
+
+
+    #kick out stuff below zero time in the trace
+    kickOutStuffBelowZero=True
```

Comment:
I really dislike this terminology. "Kick out" doesn't tell me what's going on. I suggest using more precise words like "filter" or "attenuate".